src/python/pants/engine/internals/graph.py

```diff
@@ -681,6 +681,14 @@ class _DependencyMapping:
     mapping: FrozenDict[Address, tuple[Address, ...]]
     visited: FrozenOrderedSet[Target]
     roots_as_targets: Collection[Target]
+    transitive_excludes: FrozenOrderedSet[Address]
+
+
+@dataclass(frozen=True)
+class _DependenciesInfo:
+    addresses: Addresses
+    # These are the transitive excludes gathered from dependency inference plugins
+    transitive_excludes: FrozenOrderedSet[Address]
```

Comment:
1. I don't love the name. Please suggest a better one.
2. I think we could make `transitive_excludes` include the literal transitive excludes as well, then simplify the `TransitiveTargets` rule.

Reply:
> 1. I don't love the name. Please suggest a better one.

`exclude_targets`?

> 2. I think we could make `transitive_excludes` include the literal transitive excludes as well, then simplify the `TransitiveTargets` rule.

Sounds cool. My head is not fully in the game here..